playground-core/index.js

```diff
@@ -408,6 +415,41 @@ function tdValidator(tdString, logFunc, { checkDefaults=true, checkJsonLd=true }
             }
         }
 
+        /**
+         * Warns if security Definitions has no unique keys
+         * @param {object} tdStr The TD under test as string
+         */
+        function checkSecPropUniqueness(tdStr, td) {
+
+            let result = "passed"
+            try {
+                // checking whether there are securityDefinitions at all
+                jsonValidator.parse(tdStr, false)
+            }
+            catch (error) {
+                // there is a duplicate somewhere
+
+                // convert it into string to be able to process it
+                // error is of form = Error: Syntax error: duplicated keys "overheating" near ting": {
+                const errorString = error.toString()
+                // to get the name, we need to remove the quotes around it
+                const startQuote = errorString.indexOf('"')
+                // slice to remove the part before the quote
+                const restString = errorString.slice(startQuote + 1)
+                // find where the interaction name ends
+                const endQuote = restString.indexOf('"')
+                // finally get the interaction name
+                const interactionName = restString.slice(0, endQuote)
+
+                if (td.securityDefinitions.hasOwnProperty(interactionName)) {
```

Comment:
could you change interactionName to securitySchemeName ? :)

Reply:
yes sure 😄

Reply:
https://github.com/thingweb/thingweb-playground/pull/135/commits/42ca65fa06fe70392831f889b6448bacf894c06b